### PR TITLE
Fix LZ4 file packaging support when MODULARIZE=1

### PR DESCRIPTION
**Pull request by dalcde**
This is based on #7107.

### CODE REVIEW

**Comment by dalcde** [comment]
Rebased and squashed to current master.

File: tests/test_browser.py [hunk 1]
```diff
@@ -1383,6 +1383,15 @@ def test_fs_lz4fs_package(self):
...
     self.btest(os.path.join('fs', 'test_lz4fs.cpp'), '2', args=['--pre-js', 'files.js', '-s', 'LZ4=1', '-s', 'FORCE_FILESYSTEM', '-O2'])
+    print('    modularize')
+    self.compile_btest([path_from_root('tests', 'fs', 'test_lz4fs.cpp'), '--pre-js', 'files.js', '-s', 'LZ4=1', '-s', 'FORCE_FILESYSTEM', '-s', 'MODULARIZE=1'])
+    create_test_file('a.html', '''
+      <script src="a.out.js"></script>
+      <script>
+        Module()
+      </script>
+    ''')
+    self.run_browser('a.html', '.', '/report_result?2')
```
Comment (kripken) [inline tests/test_browser.py:1394]: Is there a reason for using `compile_btest / run_browser`? I think using `btest` as in the previous cases would be simpler.

**Review by kripken — COMMENTED** [review]
Thanks! Yes, I think this is the right solution. In general we do not want to export names unnecessarily, but for this option we have to, for the filesystem integration.

Looks good aside from one comment on testing.

File: tests/test_browser.py (hunk 1 above)
Comment (dalcde) [reply tests/test_browser.py:1394]: `-s MODULARIZE` requires loading the output javascript in a different way. I'm not very familiar with the test system, but I didn't manage to get it to build with `btest`; all other tests that invoke `-s MODULARIZE` seem to do this

Comment (kripken) [reply tests/test_browser.py:1394]: Oh, right, we need a custom HTML here, thanks.

**Review by kripken — APPROVED** [review]

**Pull request merged**